### PR TITLE
Socket.io Integration

### PR DESCRIPTION
**Pull request by gint0kix**
Check It Out -Albert Lee 2018

### CODE REVIEW

**Comment by RajendraBhagroo** [comment]
Chris in server.js you used server before defining it. It's causing circle-ci to fail

**Comment by RajendraBhagroo** [comment]
Chris, chance to root directory and run [npm run circle-ci]. It says socket is not a function.

File: server/package.json [hunk 1]
```diff
@@ -17,6 +17,8 @@
     "mongoose": "^5.4.10",
     "passport": "^0.4.0",
     "passport-jwt": "^4.0.0",
+    "socket.io": "^2.2.0",
+    "socket.io-stream": "^0.9.1",
```
Comment (RajendraBhagroo) [inline server/package.json]: Is this dependency for the server or for the client?

File: server/server.js [hunk 2]
```diff
@@ -29,5 +31,44 @@ app.use(`/api/${process.env.SERVER_VERSION}/profile`, profile);
 
 // Run Server
 const host = process.env.HOST || `127.0.0.1`;
-const port = process.env.NODE_PORT || 3001;
+const port = process.env.NODE_PORT || 3009;
```
Comment (RajendraBhagroo) [inline server/server.js]: Lets try to keep NODE_PORT at 3001 in the .env file and in the server. Try changing the port Socket.io listens on to 3002

File: server/server.js [hunk 3]
```diff
@@ -29,5 +31,44 @@ app.use(`/api/${process.env.SERVER_VERSION}/profile`, profile);
...
+     */
+    socket.on("disconnect",function(data){
+      connections.splice(connections.indexOf(socket),1);
+      console.log(`Connection disconnected, number of connections ${connections.length}`);
+      socket.join("/");
+    })
+    })
+
+
+;
```
Comment (RajendraBhagroo) [inline server/server.js]: Have you tested these changes to ensure that it works properly? Also, try moving this code into its own file and including it in here. Like I did with profile and users, if possible

File: server/package.json (hunk 1 above)
Comment (gint0kix) [reply server/package.json]: for the server

File: server/server.js (hunk 3 above)
Comment (gint0kix) [reply server/server.js]: yeah it works, don't believe its possible, all tutorials have it running on the server js

**Comment by RajendraBhagroo** [comment]
Chris, if you added a new variable to our private [.env] file, please add it to the public [.env.example] and update the corresponding code in the repositories readme.md so other developers know to add this variable to their private [.env] files.

Thank you! I will add the variable on circleCI so hopefully your tests stop failing, though they shouldn't fail if you kept express on the same port and move socket.io on another port

File: client/src/components/layout/Navbar.js [hunk 4]
```diff
@@ -28,6 +28,12 @@ class Navbar extends React.Component {
 
     const authLinks = (
       <ul className="navbar-nav ml-auto">
+        <li className="nav-item">
+          <Link to="/streams" className="nav-link">
+            Streams
+            <i className="fas fa-sign-out-alt ml-2" />
```
Comment (RajendraBhagroo) [inline client/src/components/layout/Navbar.js]: Remember To Change Icon From Sign Out To Something That Resembles Streams. We Use FontAwsome For Icons, Check Out Their Website https://fontawesome.com/

File: client/src/components/stream/Stream.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+
+class Stream extends Component{
+    /*Upon mounting of stream, the user's video, and audio will be obtained
```
Comment (RajendraBhagroo) [inline client/src/components/stream/Stream.js]: See Other Components To Declare State. You Set State On Line 23 For [videoSrc], However Compononet Level State Is Not Defined And [videoSrc] Does Not Exist.

File: server/server.js [hunk 6]
```diff
@@ -2,10 +2,11 @@ const express = require("express");
 const mongoose = require("mongoose");
 const passport = require("passport");
 const cors = require("cors");
-
 const app = express();
 require(`./${process.env.SERVER_VERSION}/passport`)(passport);
-
+var server = require('http').Server(app);
```
Comment (RajendraBhagroo) [inline server/server.js]: Change var To let

File: server/server.js [hunk 7]
```diff
@@ -2,10 +2,11 @@ const express = require("express");
...
+var io = require('socket.io').listen(server)
```
Comment (RajendraBhagroo) [inline server/server.js]: Change [var] to [let]

**Review by RajendraBhagroo — CHANGES_REQUESTED** [review]

**Comment by RajendraBhagroo** [comment]
@gint0kix  Nice Work!

I was looking over the code and I think we should try to meet before Operating Systems Security to discuss the new code. The Stream start seems to respond, As well as stream end. Video start and end are good. Recording start and stop throw errors for media player. There are also many unused variables.

image

image

**Review by RajendraBhagroo — APPROVED** [review]

**Review by Gold-Turtle — APPROVED** [review]
Approved

**Comment by RajendraBhagroo** [comment]
The ICE Issue with RTC Peer Connection Is A False Positive. ICE does not handle stream, Googles STUN servers will. Accepted !

**Pull request closed without merging**